Examples/Algorithms/Geant4/src/EventAction.cpp

```diff
@@ -65,7 +65,8 @@ void EventAction::clear() {
 ///
 /// This only contains valid data after the end-of-event action has been
 /// executed.
-const std::vector<Acts::RecordedMaterialTrack>& EventAction::materialTracks()
-    const {
-  return m_materialTracks;
+std::vector<Acts::RecordedMaterialTrack> EventAction::materialTracks() {
```

Comment:
I think this is misleading. If this function is a just an accessor (as the naming suggest) then it should not modify the class. Or is modifies the class, in which case the function should be named differently. My preference would be to follow [Command-query-separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) here, i.e. leave `materialTracks()` as a const accessors and add a separate `clearMaterialTracks()` method that is called at the appropriate point.

Reply:
It's a bit of a problem with how we have to interact with Geant4.

What we can do is:
- keep the checkout const (as it was)
- move the clear() statement to the BeginOfEvent() action ?

@Corentin-Allaire - would that work?

I think that's the Geant4 way of doing this (not sure how they do MT then though).

Reply:
If you want to keep it in a single function call, I would suggest to rename it e.g. to `extractMaterialTracks()` to make clear what it does. I would prefer have separate accessors and clear class, but I understand that might not work with Geant4.

Reply:
I 100% agree with the renaming. We could also move the clear to the begin of event, in particular now that I look at it we have a EventAction::clear() method that is already called at the BeginOfEventAction() but that is empty. I guess we could simply move the m_materialTracks.clear() into the EventAction::clear() ?

Reply:
I think that would be the `clear` est! I Let's move it to `EventAction::clear()`

Reply:
The reason for why I did it that way is that since I don't fully understand G4 I thought we might have had a reason for doing it that way in the past... But maybe not.

Reply:
I still have one issue when writing the materialTracks to the event store in GeantinoRecording.cpp, if I write a const object I don't appear to be able to read it back afterward. Am I missing something ? That was the reason why I switch EventAction::materialTracks() away from a const to beging with.

Reply:
It could be that the whiteboard gets confused with the types and stores the reference instead of the object itself. That is just a guess though. I would still return a const references from the accessor but create a copy manually before you write it to the whiteboard, i.e.

```cpp
auto mt = action.materialTracks();
eventStore.add(..., std::move(mt));
```

Reply:
Looking at how WriterT is templated that was also my guess, I can definetly move the copy outside the EventAction::materialTracks(). Conceder it done !